packages/docusaurus-types/src/index.d.ts

```diff
@@ -5,784 +5,72 @@
- * @see https://help.phrase.com/help/chrome-json-messages
- */
-export type TranslationMessage = {message: string; description?: string};
-export type TranslationFileContent = {[msgId: string]: TranslationMessage};
-/**
- * An abstract representation of how a translation file exists on disk. The core
- * would handle the file reading/writing; plugins just need to deal with
- * translations in-memory.
- */
-export type TranslationFile = {
-  /**
-   * Relative to the directory where it's expected to be found. For plugin
-   * files, it's relative to `i18n/<locale>/<pluginName>/<pluginId>`. Should NOT
-   * have any extension.
-   */
-  path: string;
-  content: TranslationFileContent;
-};
-
-export type I18n = DeepRequired<I18nConfig> & {currentLocale: string};
-
-export type GlobalData = {[pluginName: string]: {[pluginId: string]: unknown}};
-
-export type CodeTranslations = {[msgId: string]: string};
-
-export type DocusaurusContext = {
-  siteConfig: DocusaurusConfig;
-  siteMetadata: SiteMetadata;
-  globalData: GlobalData;
-  i18n: I18n;
-  codeTranslations: CodeTranslations;
-
-  // Don't put mutable values here, to avoid triggering re-renders
-  // We could reconsider that choice if context selectors are implemented
-  // isBrowser: boolean; // Not here on purpose!
-};
-
-export type LoadContext = {
-  siteDir: string;
-  generatedFilesDir: string;
-  siteConfig: DocusaurusConfig;
-  siteConfigPath: string;
-  outDir: string;
-  /**
-   * Directory where all source translations for the current locale can be found
-   * in. Constructed with `i18n.path` + `i18n.currentLocale.path` (e.g.
-   * `<siteDir>/i18n/en`)
-   */
-  localizationDir: string;
-  /**
-   * Duplicated from `siteConfig.baseUrl`, but probably worth keeping. We mutate
-   * `siteConfig` to make `baseUrl` there localized as well, but that's mostly
-   * for client-side. `context.baseUrl` is still more convenient for plugins.
-   */
-  baseUrl: string;
-  i18n: I18n;
-  codeTranslations: CodeTranslations;
-};
-
-export type Props = LoadContext & {
-  headTags: string;
-  preBodyTags: string;
-  postBodyTags: string;
-  siteMetadata: SiteMetadata;
-  routes: RouteConfig[];
-  routesPaths: string[];
-  plugins: LoadedPlugin[];
-};
-
-// === Plugin ===
-
-export type PluginContentLoadedActions = {
-  addRoute: (config: RouteConfig) => void;
-  createData: (name: string, data: string) => Promise<string>;
-  setGlobalData: (data: unknown) => void;
-};
-
-export type ConfigureWebpackUtils = {
-  getStyleLoaders: (
-    isServer: boolean,
-    cssOptions: {[key: string]: unknown},
-  ) => RuleSetRule[];
-  getJSLoader: (options: {
-    isServer: boolean;
-    babelOptions?: {[key: string]: unknown};
-  }) => RuleSetRule;
-};
-
-export type AllContent = {
-  [pluginName: string]: {
-    [pluginID: string]: unknown;
-  };
-};
-
-// TODO improve type (not exposed by postcss-loader)
-export type PostCssOptions = {plugins: unknown[]; [key: string]: unknown};
-
-type HtmlTagObject = {
```

Comment:
not sure why: this type was not exported from this package and yet it was imported successfully in Docusaurus core 🤷‍♂️

Reply:
Because this is a d.ts file with no value export, so it's still ambient. We can fix this by adding an `export {}`.

Reply:
Ah I see 😅 

Not very familar with those TS rules and value exports, where is this documented? Should we add `export {}` to all those `.d.ts` files?